src/lazy/text/buffer.rs

```diff
@@ -575,20 +602,19 @@ impl<'data> TextBufferView<'data> {
     /// * An identifier
     /// * A symbol ID
     /// * A short-form string
-    pub fn match_struct_field_name(self) -> IonParseResult<'data, MatchedFieldName> {
-        alt((
-            Self::match_string.map(MatchedFieldName::String),
-            Self::match_symbol.map(MatchedFieldName::Symbol),
-        ))(self)
+    pub fn match_struct_field_name(self) -> IonParseResult<'top, MatchedFieldName> {
+        match_and_span(alt((
```

Comment:
🗺️ Previously, every call to this method would wrap it in a `match_and_span(` call. Now the method performs that step itself, simplifying a few call sites.